### DISCUSSION TITLE
How do I add indentation to code blocks?

### DISCUSSION THREAD

**Discussion by Calab** [TriliumNext/Trilium#4693] [Q&A] [answered]
Something  I just noticed in Trilium is that you cannot indent within a code block. That severly limits the usefulness of a code block. Python, for example, depends on code indentation. Not being able do put indents in the code make Python code blocks impossible to use.

Am I just being stupid and overlooking something, or is this a real limitation?

![Trilium](https://github.com/zadam/trilium/assets/316725/9666c1d2-aa00-4684-8534-84afadb5e372)

**Comment by meichthys** [comment]
This is not currently possible due to ckeditor limitations. There may be a way to work around this using css, but it would be hacky.

If you just want to add spaces before the line, you should be able to do that without issue.

**Reply by Calab** [reply, marked as answer]
Spaces and tabs are both ignored. I ended up using ALT-255 to enter some blank non-space characters.

It looks like something I did with the CSS...

code{  white-space: nowrap !important;   }

...is hiding the leading spaces in my code blocks. Changing from nowrap to pre seems to have fixed the spacing issue.

Thanks for letting me know that leading spaces should have been working.
